### PR TITLE
Prelude : Bump version 3.1.0

### PR DESCRIPTION
**Pull request by ToToL**
New release on 14 september 2016

### CODE REVIEW

**Comment by ToToL** [comment]
https://bugs.gentoo.org/show_bug.cgi?id=600372

File: dev-libs/libpreludedb/libpreludedb-3.1.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+	python? ( ${PYTHON_DEPS} )
+	mysql? ( virtual/libmysqlclient )
+	postgres? ( dev-db/postgresql:* )
+	sqlite? ( dev-db/sqlite:3 )"
+
+DEPEND="${RDEPEND}
+	doc? ( dev-util/gtk-doc )
+	sys-devel/flex
+	virtual/yacc
+	>=dev-lang/swig-3.0.7
```
Comment (gktrk) [inline dev-libs/libpreludedb/libpreludedb-3.1.0.ebuild]: Should we update swig version range? https://github.com/Prelude-SIEM/libpreludedb/compare/3.0.1...3.1.0#diff-cbbb15035ffe792595354338f7196db9R3

File: app-admin/prelude-manager/files/prelude-manager-3.1.0-run.patch [hunk 2]
```diff
@@ -0,0 +1,13 @@
+When : everytime
+Why : since systemd, run dir has changed
```
Comment (gktrk) [inline app-admin/prelude-manager/files/prelude-manager-3.1.0-run.patch]: I am not sure if this information helps much. Let's just change it to a single non-awkward sentence: `Fix the run dir to comply with the Gentoo FS layout.`

File: dev-libs/libprelude/libprelude-3.1.0.ebuild [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+	perl? ( dev-lang/perl virtual/perl-ExtUtils-MakeMaker )
+	lua? ( dev-lang/lua:* )
+	ruby? ( ${RUBY_DEPS} )
+	python? ( ${PYTHON_DEPS} )"
+
+DEPEND="${RDEPEND}
+	doc? ( dev-util/gtk-doc )
+	sys-devel/flex
+	virtual/yacc
+	>=dev-lang/swig-3.0.7
```
Comment (ToToL) [inline dev-libs/libprelude/libprelude-3.1.0.ebuild]: We provide swig files if the system does not provide it. if not, swig >= 3.0.7 is required, this range do not change. 3.0.7 fix an issue with python3.5

File: www-apps/prewikka/prewikka-3.1.0.ebuild [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+DESCRIPTION="Graphical front-end analysis console for the Prelude Framework"
+HOMEPAGE="https://www.prelude-siem.org"
+SRC_URI="https://www.prelude-siem.org/pkg/src/${PV}/${P}.tar.gz"
+
+LICENSE="GPL-2+"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="adns"
+
```
Comment (ToToL) [inline www-apps/prewikka/prewikka-3.1.0.ebuild]: OK

File: net-analyzer/prelude-lml/prelude-lml-3.1.0.ebuild [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+LICENSE="GPL-2+"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="tls icu"
+
+RDEPEND="dev-libs/libprelude
+	dev-libs/libpcre
+	icu? ( dev-libs/icu )
+	tls? ( net-libs/gnutls )"
```
Comment (ToToL) [inline net-analyzer/prelude-lml/prelude-lml-3.1.0.ebuild]: I have to check but it should work ;)

File: app-admin/prelude-manager/files/prelude-manager-3.1.0-run.patch (hunk 2 above)
Comment (ToToL) [inline app-admin/prelude-manager/files/prelude-manager-3.1.0-run.patch]: ok

**Comment by kentfredric** [comment]
I can't tell by merely looking at this PR, but if `USE="perl"` either:

1. Links to `libperl.so.*`
2. Installs `.pm` files in a path with a version like that found in `perl -V:version`

Then the `RDEPEND` should contain 

 ```bash
dev-lang/perl:=
```

In order to force this package to recompile with major perl upgrades.

File: app-admin/prelude-manager/prelude-manager-3.1.0.ebuild [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+
+src_prepare() {
+	default_src_prepare
+
+	eautoreconf
+}
+
+src_configure() {
+	econf \
+		--localstatedir=/var \
```
Comment (SoapGentoo) [inline app-admin/prelude-manager/prelude-manager-3.1.0.ebuild]: `${EPREFIX}/var`

File: app-admin/prelude-manager/prelude-manager-3.1.0.ebuild [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+
+DEPEND="${RDEPEND}
+	virtual/pkgconfig"
+
+PATCHES=(
+	"${FILESDIR}/${P}-run.patch"
+)
+
+src_prepare() {
+	default_src_prepare
```
Comment (SoapGentoo) [inline app-admin/prelude-manager/prelude-manager-3.1.0.ebuild]: just say `default`

File: app-admin/prelude-manager/prelude-manager-3.1.0.ebuild [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+	econf \
+		--localstatedir=/var \
+		$(use_with dbx libpreludedb-prefix) \
+		$(use_with tcpwrapper libwrap) \
+		$(use_with xml xml-prefix) \
+		$(use_enable geoip libmaxminddb)
+}
+
+src_install() {
+	default_src_install
```
Comment (SoapGentoo) [inline app-admin/prelude-manager/prelude-manager-3.1.0.ebuild]: `default`

File: app-admin/prelude-manager/prelude-manager-3.1.0.ebuild [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+
+	rm -rv "${D}/run" || die "rm failed"
```
Comment (SoapGentoo) [inline app-admin/prelude-manager/prelude-manager-3.1.0.ebuild]: `rm -rv "${ED%/}/run"`

File: app-admin/prelude-manager/prelude-manager-3.1.0.ebuild [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+	keepdir /var/spool/prelude-manager{,/failover,/scheduler}
+
+	prune_libtool_files --modules
```
Comment (SoapGentoo) [inline app-admin/prelude-manager/prelude-manager-3.1.0.ebuild]: Can you check these by hand? https://wiki.gentoo.org/wiki/Project:Quality_Assurance/Handling_Libtool_Archives

File: dev-libs/libprelude/libprelude-3.1.0.ebuild [hunk 11]
```diff
@@ -0,0 +1,77 @@
...
+
+LICENSE="GPL-2+"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="doc lua python ruby perl"
+
+REQUIRED_USE="python? ( ${PYTHON_REQUIRED_USE} )"
+
+RDEPEND="dev-libs/libgcrypt:0=
+	net-libs/gnutls
```
Comment (SoapGentoo) [inline dev-libs/libprelude/libprelude-3.1.0.ebuild]: needs `:=`

File: net-analyzer/prelude-lml/prelude-lml-3.1.0.ebuild (hunk 5 above)
Comment (SoapGentoo) [inline net-analyzer/prelude-lml/prelude-lml-3.1.0.ebuild]: also `USE=tls` sounds wrong, shouldn't this be `USE=ssl`? Also needs `:=`

**Comment by ToToL** [comment]
If I missed nothing, it is now OK

Thanks for the review

**Comment by ToToL** [comment]
I fixed QA issue

File: app-admin/prelude-manager/prelude-manager-3.1.0.ebuild [hunk 12]
```diff
@@ -0,0 +1,67 @@
...
+RDEPEND="net-libs/gnutls:=
+	dev-libs/libprelude
+	dbx? ( dev-libs/libpreludedb )
+	tcpwrapper? ( sys-apps/tcp-wrappers )
+	xml? ( dev-libs/libxml2 )
+	geoip? ( dev-libs/libmaxminddb )"
+
+DEPEND="${RDEPEND}
+	virtual/pkgconfig
+	libev? ( dev-libs/libev )"
```
Comment (gktrk) [inline app-admin/prelude-manager/prelude-manager-3.1.0.ebuild]: Is `libev` is a build time dependency and not a runtime dependency? I did an objdump on prelude-manager and it seems like there are references to `ev_*` functions.

File: app-admin/prelude-manager/metadata.xml [hunk 13]
```diff
@@ -24,5 +24,7 @@
 		<flag name="dbx">Enables Prelude Manager storing Prelude
 		alerts into database through <pkg>dev-libs/libpreludedb</pkg>
 		.</flag>
+        <flag name="libev">Support event loops using <pkg>dev-libs/libev</pkg>
```
Comment (gktrk) [inline app-admin/prelude-manager/metadata.xml]: Per https://github.com/Prelude-SIEM/prelude-manager/blob/master/configure.ac#L242 it doesn't look like libev is optional. Am I interpreting it wrong? If that's the case, this description is confusing.

If `libev` is not optional, let's not provide support for embedded libev and unconditionally use the system one.

File: app-admin/prelude-manager/prelude-manager-3.1.0.ebuild [hunk 14]
```diff
@@ -0,0 +1,67 @@
+# Copyright 1999-2016 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+# $Id$
+
+EAPI=6
+
+inherit autotools eutils systemd
```
Comment (gktrk) [inline app-admin/prelude-manager/prelude-manager-3.1.0.ebuild]: If we are deleting la files using `find`, we don't need to inherit `eutils` anymore.

File: dev-libs/libprelude/libprelude-3.1.0.ebuild [hunk 15]
```diff
@@ -0,0 +1,77 @@
...
+# Distributed under the terms of the GNU General Public License v2
+# $Id$
+
+EAPI=6
+
+PYTHON_COMPAT=( python{2_7,3_{4,5}} )
+
+USE_RUBY="ruby21"
+
+inherit autotools eutils python-r1 ruby-single
```
Comment (gktrk) [inline dev-libs/libprelude/libprelude-3.1.0.ebuild]: Don't need to inherit `eutils`

File: dev-libs/libpreludedb/libpreludedb-3.1.0.ebuild [hunk 16]
```diff
@@ -0,0 +1,73 @@
+# Copyright 1999-2016 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+# $Id$
+
+EAPI=6
+
+PYTHON_COMPAT=( python{2_7,3_{4,5}} )
+
+inherit autotools eutils python-r1
```
Comment (gktrk) [inline dev-libs/libpreludedb/libpreludedb-3.1.0.ebuild]: Don't need to inherit `eutils`

File: net-analyzer/prelude-lml/prelude-lml-3.1.0.ebuild [hunk 17]
```diff
@@ -0,0 +1,62 @@
+# Copyright 1999-2016 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+# $Id$
+
+EAPI=6
+
+inherit autotools eutils systemd
```
Comment (gktrk) [inline net-analyzer/prelude-lml/prelude-lml-3.1.0.ebuild]: No need to inherit `eutils`

**Comment by ToToL** [comment]
New version

Note : file mode changes are intentional because of QA

**Comment by gktrk** [comment]
prelude-lml failed to compile for me. looks like a libev issue. I'll investigate this tonight. here's the log if you want to take a look at it: https://dev.gentoo.org/~gokturk/proxy-maint/pr2938/net-analyzer:prelude-lml-3.1.0:20161206-005431.log

File: app-admin/prelude-manager/prelude-manager-3.1.0.ebuild [hunk 18]
```diff
@@ -0,0 +1,62 @@
...
+HOMEPAGE="https://www.prelude-siem.org"
+SRC_URI="https://www.prelude-siem.org/pkg/src/${PV}/${P}.tar.gz"
+
+LICENSE="GPL-2+"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="tcpwrapper xml geoip dbx"
+
+RDEPEND="net-libs/gnutls:=
+	dev-libs/libprelude
```
Comment (gktrk) [inline app-admin/prelude-manager/prelude-manager-3.1.0.ebuild]: Per https://github.com/Prelude-SIEM/prelude-manager/blob/master/configure.ac#L50 we need `>=dev-libs/libprelude-3.1.0`

File: dev-libs/libpreludedb/libpreludedb-3.1.0.ebuild [hunk 19]
```diff
@@ -0,0 +1,73 @@
...
+LICENSE="GPL-2+"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="doc python mysql postgres sqlite"
+
+REQUIRED_USE="python? ( ${PYTHON_REQUIRED_USE} )"
+
+RDEPEND="dev-libs/libgcrypt:0=
+	net-libs/gnutls:=
+	dev-libs/libprelude
```
Comment (gktrk) [inline dev-libs/libpreludedb/libpreludedb-3.1.0.ebuild]: Per https://github.com/Prelude-SIEM/libpreludedb/blob/master/configure.ac#L135 we need `>=dev-libs/libprelude-3.1.0` here

File: net-analyzer/prelude-correlator/prelude-correlator-3.1.0.ebuild [hunk 20]
```diff
@@ -0,0 +1,31 @@
...
+SRC_URI="https://www.prelude-siem.org/pkg/src/${PV}/${P}.tar.gz"
+
+LICENSE="GPL-2+"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+
+DEPEND="dev-python/setuptools"
+
+RDEPEND="dev-python/netaddr[${PYTHON_USEDEP}]
+	dev-libs/libprelude[python,${PYTHON_USEDEP}]"
```
Comment (gktrk) [inline net-analyzer/prelude-correlator/prelude-correlator-3.1.0.ebuild]: Per https://github.com/Prelude-SIEM/prelude-correlator/blob/master/setup.py#L155 we need `>=dev-libs/libprelude-3.1.0` here

**Comment by gktrk** [comment]
I tried to compile prelude-lml with all the libev versions and they all fail. I didn't expect a divergence from the upstream. What do you suggest that we should do, just switch to the embedded one?

**Comment by ToToL** [comment]
Anyway, I have to patch for libev, let it be embedded in this version and hope we can use libev from system in the next version

**Comment by gktrk** [comment]
You're the upstream, sounds fine by me.

**Comment by gktrk** [comment]
Merged in https://github.com/gentoo/gentoo/compare/30f4dcd...6772e82. Thanks!

**Pull request closed without merging**